traits/tests/test_callable.py

```diff
@@ -0,0 +1,69 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+import unittest
+
+from traits.api import Callable, HasTraits, TraitError
+
+
+
+def function():
+    pass
+
+
+class Dummy(object):
+
+    def instance_method(self):
+        pass
+
+
+class MyCallable(HasTraits):
+
+    value = Callable()
+
+
+class MyCallableCanBeNone(HasTraits):
+
+    value = Callable(allow_none=True)
+
+
+class TestCallable(unittest.TestCase):
+
+    def test_default(self):
+        a = MyCallable()
+        self.assertIsNone(a.value)
+
+    def test_accepts_lambda(self):
+        func = lambda v: v + 1
+        a = MyCallable(value=func)
+        self.assertIs(a.value, func)
+
+    def test_accepts_type(self):
+        MyCallable(value=float)
+
+    def test_accepts_method(self):
+        dummy = Dummy()
+        MyCallable(value=dummy.instance_method)
+
+    def test_accepts_none(self):
```

Comment:
Would it be worth adding a test that if `allow_none=False` (or is unspecified), then `None` is *not* accepted?

Reply:
I added that in the first commit, and that did not pass, indicating that this feature was not supported on master?  So I removed it in the second commit.
```
======================================================================
FAIL: test_rejects_none (traits.tests.test_callable.TestCallable)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kit/git/traits/traits/tests/test_callable.py", line 78, in test_rejects_none
    MyCallableCannotBeNone(value=None)
AssertionError: TraitError not raised

```